### DISCUSSION TITLE
How to get a drawable size of the window?

### DISCUSSION THREAD

**Discussion by dnrusakov** [hecrj/iced#1401] [Q&A] [answered]
1. How to get a drawable size of the window?
2. How to get a drawable size of the window inside `update()` method?

**Comment by dnrusakov** [comment]
cc @hecrj

**Comment by tarkah** [comment, marked as answer]
You should be using [iced_lazy::responsive](https://docs.iced.rs/iced_lazy/fn.responsive.html) which gives you the `Size` of the widget. If you wrap your root widget inside this during `view`, you will effectively get the viewport size of your window.

If you need it during `update`, the best way currently would be to have a `RefCell<Size>` / `Size` (depending if you're using `pure` or not and have `&self` / `&mut self` in view) that you update during `view` with the `Size` provided by the responsive widget.

**Reply by dnrusakov** [reply]
@tarkah Thank you!
